app/repositories/vector_repository.py

```
@@ -0,0 +1,312 @@
+        """
+        base_similarity = (1 - DocumentVector.embedding.cosine_distance(query_embedding))
+        
+        # 키워드 매칭 점수 계산
+        keyword_score = 0.0
+        if query_text:
+            # PostgreSQL의 텍스트 검색 기능 사용
+            keyword_score = text("ts_rank(to_tsvector('english', content), to_tsquery('english', :query_text))").bindparam(query_text=query_text)
+        
+        if keywords:
+            # JSONB 키워드 배열에서 매칭되는 키워드 수에 따른 가중치
+            keyword_match_score = text("jsonb_array_length(keywords) * 0.1").label("keyword_match_score")
```

Ruff (0.11.9)

Local variable `keyword_match_score` is assigned to but never used

Remove assignment to unused variable `keyword_match_score`

(F841)